packages/create-svelte/test/check.js

```diff
@@ -89,7 +89,11 @@ for (const template of fs.readdirSync('templates')) {
 				const command = `pnpm run ${script}`;
 				try {
 					console.log(`executing ${command} in ${cwd}`);
-					execSync(command, { cwd, stdio: 'inherit' });
+					execSync(command, {
+						cwd,
+						stdio: 'inherit',
+						env: { ...process.env, GITHUB_ACTIONS: undefined }
```

Comment:
I think the fact that you have to do this tells us that checking for `GITHUB_ACTIONS` is too broad. Is there an environment variable more specific to GitHub pages that we can look for?

Reply:
Totally agree.

I don't see anything related to GitHub Pages in [Default environment variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables).

Traditionally, GitHub Actions that would publish on GitHub Pages don't have anything that would differentiate them:

```yaml
      - name: Build
        run: pnpm run build
      - name: GitHub Pages
        run: |
          # Setup Git
          cd build
          git init --initial-branch gh-pages
          git remote add origin git@github.com:sveltejs/kit.git
          git config user.name 'GitHub Pages'
          git config user.email '<>'
          eval `ssh-agent -s`
          ssh-add - <<< '${{ secrets.GH_PAGES_SSH }}'

          # Publish
          git add .
          git commit -m "pnpm run build"
          git push --set-upstream origin gh-pages -f
```

It would be possible to make up an arbitrary environment variable `GITHUB_PAGES` and use it, but people would need to set it manually like `GITHUB_PAGES=true pnpm run build` and it doesn't look as nice.

I did find a public beta for [actions/upload-pages-artifact](https://github.com/actions/upload-pages-artifact) and [actions/deploy-pages](https://github.com/actions/deploy-pages) and some docs at [Creating a custom GitHub Actions workflow to publish your site](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site), but I don't think these set an environment variable aside from `GITHUB_ACTION_REPOSITORY`. Plus, by the time the variable is set, the build would have been done already.

Reply:
A `$GITHUB_PAGES` has been proposed in https://github.com/actions/configure-pages/pull/16. It would require `actions/configure-pages` to be ran before the build, but I think it's an elegant solution.

```yaml
      - uses: actions/configure-pages@v1
      - name: Build
        run: pnpm run build
      - uses: actions/upload-pages-artifact@v1
        with:
          path: build
      - uses: actions/deploy-pages@v1
```

Reply:
`$GITHUB_PAGES` was [approved](https://github.com/actions/configure-pages/pull/16#discussion_r946199413)